maven-plugin/src/main/java/org/openmrs/maven/plugins/Run.java

```diff
@@ -73,13 +81,113 @@ public void executeTask() throws MojoExecutionException, MojoFailureException {
 			}
 		}
 
+		try {
+			File tempFolder = createTempReactorProject(server);
+			cleanInstallModule(tempFolder.getPath());
+			deleteTempReactorProject(tempFolder);
+			deployWatchedModules(server);
+		} catch (MavenInvocationException e) {
+			throw new MojoFailureException(e.getMessage());
+		}
+
 		if (Boolean.FALSE.equals(fork)) {
 			new RunTomcat(serverId, port, wizard).execute();
 		} else {
 			runInFork(server);
 		}
 	}
 
+	private File createTempReactorProject(Server server) throws MojoFailureException, MojoExecutionException {
+		File tempFolder = new File(server.getServerDirectory(), "temp-project");
+		if (!tempFolder.exists()) {
+			tempFolder.mkdir();
+		}
+		Model tempModel = createModel();
+		Set<Project> watchedModules = server.getWatchedProjects();
+		for(Project module: watchedModules){
+			File srcDir = new File(module.getPath());
+			File moduleFile = new File(tempFolder, module.getArtifactId());
+			try {
+				FileUtils.copyDirectory(srcDir, moduleFile);
+				Model model = Project.loadProject(moduleFile).getModel();
+				Parent parent = new Parent();
+				parent.setArtifactId("temp-project");
+				parent.setGroupId("org.temp");
+				parent.setVersion("1.0");
+				model.setParent(parent);
```

Comment:
We should not be setting parents. It should be enough to add them as modules to the temp-project.